backend/data.js

```diff
@@ -80,11 +81,39 @@ const getCardByUuid = (uuid) => {
   return getCards()[uuid];
 };
 
-const getCubableCardByName = (cardName) => {
+const parseCubableCardName = (cardName) => {
+  // Cube cards can be written as:
+  //
+  // * "Abrade" (just card name)
+  // * "Abrade (CMR)" (card name + set code)
+  // * "Abrade (CMR 410)" (card name + set code + number within set)
+  const match = cardName.match(/^(.*?)(?: +\(([a-z0-9]+)(?: +([a-z0-9]+))?\))? *$/);
```

Comment:
Looks like your regex doesn't match what it should be matching. you forgot about capitals in your set code and card number. The fix would be `[A-Za-z0-9]+`, but you can shorthand that to simply `\w+`, which is what i did in my replacement.

```suggestion
  const match = cardName.match(/^(.*?)(?: +\((\w+)(?: +(\w+))?\))? *$/);
```

Reply:
The card names are all lowercased before they're passed here, but you're right that `\w+` is more obvious and more terse.